### DISCUSSION TITLE
search string if it is invalid link

### DISCUSSION THREAD

**Discussion by shalva97** [minbrowser/min#1658] [Ideas]
okay so I have this string `android.arch.lifecycle:extensions:2.1.0` it is a library for android. All other browsers, Edge, Firefox and Chrome directly search it in Bing.

it would be better if Min did same, it can already detect if url is invalid so this functionality should be easy to add.

**Comment by lubiedo** [comment]
Thanks for starting this discussion. The string in question and `any.string.like.this` will be interpreted as a domain name. I was thinking that we could use a library to parse the domains and validate (ie. https://github.com/peerigon/parse-domain) in urlParser.

What do you think, @PalmerAL?

**Reply by PalmerAL** [reply]
Yeah, that would be the right way to do it. I think using that library is fine, as long as parsing time / memory usage is good (I'm assuming it probably is, but I don't see a benchmark anywhere).

**Reply by lubiedo** [reply]
Created a PR: https://github.com/minbrowser/min/pull/1667
Let me know what you think. Might be something more simple than using another package.
